app/src/main/java/com/github/siela1915/bootcamp/AutocompleteApi/IngredientAutocomplete.java

```
@@ -94,38 +110,84 @@ public void onFailure(Call<List<ApiResponse>> call, Throwable t) {
             @Override
             public void onResponse(Call<NutrientsResponse> call, Response<NutrientsResponse> response) {
                 if(response.isSuccessful()) {
                     if (response.body() != null) {
-                        for (Nutrient nut : response.body().nutrients) {
-                            switch(nut.name){
+                        Map<String, Double> map = response.body().nutrition.mapFromNutrients();
+                        for(String s : Nutrition.NUTRIENT_NAMES){
+                            switch (s) {
+                                //map.get() won't return null due to how the map is constructed in the Nutrition class
                                 case "Calories":
-                                    ingredient.setCalories(nut.amount);
+                                    ingredient.setCalories(map.get(s));
                                     break;
                                 case "Fat":
-                                    ingredient.setFat(nut.amount);
-                                    break;
-                                case "Carbohydrates":
-                                    ingredient.setCarbs(nut.amount);
+                                    ingredient.setFat(map.get(s));
                                     break;
                                 case "Sugar":
-                                    ingredient.setSugar(nut.amount);
+                                    ingredient.setSugar(map.get(s));
                                     break;
                                 case "Protein":
-                                    ingredient.setProtein(nut.amount);
+                                    ingredient.setProtein(map.get(s));
+                                    break;
+                                case "Carbohydrates":
+                                    ingredient.setCarbs(map.get(s));
+                                    break;
+                                default:
```

You might want to throw an exception in the default case in case something goes wrong in the construction of the nutrients.